packages/gatekeeper/src/components/logout.tsx

```diff
@@ -3,39 +3,41 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { Redirect } from 'react-router-dom';
 import { LOGIN_URL } from '../helpers/constants';
-import { logoutFromAuthServer } from '../helpers/utils';
+
+export type LogoutFunction = (props: LogoutProps) => null | JSX.Element;
 
 /** interface to describe props for Logout component
  * @member {typeof logOutUser}logoutActionCreator action creator that logs out user.
  * @member {string} redirectPath The URL we redirect to after logging out.
- * @member {string} logoutURL the url of the logout endpoint of the Oauth server.
- * @member {(logoutUrl: string) => void} logoutFunction custom function to log user out of the Oauth server.
+ * @member {LogoutFunction} logoutFunction custom function to log user out of the Oauth server
+ *  - called after logging the user out of the store.
  */
 export interface LogoutProps {
   logoutActionCreator: typeof logOutUser;
   redirectPath: string;
-  logoutURL: string | null;
-  logoutFunction: (logoutUrl: string) => void | null;
+  logoutFunction: LogoutFunction;
 }
 
+/** the default logout function : redirects to the predefined redirectPath
+ * @param logoutProps - logout component props
+ */
+export const defaultLogout = (logOutProps: LogoutProps) => {
+  const { redirectPath, logoutActionCreator } = logOutProps;
+  logoutActionCreator();
+  return <Redirect to={redirectPath} />;
+};
+
 /** default props for Logout component */
 export const defaultLogoutProps: LogoutProps = {
   logoutActionCreator: logOutUser,
-  logoutFunction: logoutFromAuthServer,
-  logoutURL: null,
+  logoutFunction: defaultLogout,
   redirectPath: LOGIN_URL
 };
 
 /** Logout component */
 const Logout = (props: LogoutProps) => {
-  const { logoutActionCreator, redirectPath } = props;
-  logoutActionCreator();
-  if (props.logoutURL && props.logoutFunction) {
-    props.logoutFunction(props.logoutURL);
-  } else if (props.logoutURL) {
-    logoutFromAuthServer(props.logoutURL);
-  }
-  return <Redirect to={redirectPath} />;
+  const { logoutFunction } = props;
+  return logoutFunction(props);
```

Comment:
do we need to call it with the entire props object?

Reply:
thought it best for maximum flexibility.

Reply:
I feel weird about `logoutFunction` being passed to itself in the props.

Reply:
I guess its fine though